File: system-test/dotnet/Driver.Adapter/External/Erp/ErpRealDriver.cs
using Common;
using Driver.Port.External.Erp.Dtos;
using Driver.Port.External.Erp.Dtos.Error;
using Driver.Adapter.External.Erp.Client;
using Driver.Adapter.External.Erp.Client.Dtos;
using Driver.Adapter.External.Erp.Client.Dtos.Error;

namespace Driver.Adapter.External.Erp;

public class ErpRealDriver : BaseErpDriver<ErpRealClient>
{
    private const string DefaultTitle = "Test Product Title";
    private const string DefaultDescription = "Test Product Description";
    private const string DefaultCategory = "Test Category";
    private const string DefaultBrand = "Test Brand";

    public ErpRealDriver(string baseUrl) : base(new ErpRealClient(baseUrl))
    {
    }

    public override Task<Result<VoidValue, ErpErrorResponse>> ReturnsProductAsync(ReturnsProductRequest request)
    {
        var createProductRequest = new ExtCreateProductRequest
        {
            Id = request.Sku,
            Title = DefaultTitle,
            Description = DefaultDescription,
            Category = DefaultCategory,
            Brand = DefaultBrand,
            Price = request.Price
        };

        return _client.CreateProductAsync(createProductRequest)
            .MapErrorAsync(error => MapError(error));
    }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/adapter/external/erp/ErpRealDriver.java
package com.optivem.shop.dsl.driver.adapter.external.erp;

import com.optivem.shop.dsl.driver.adapter.external.erp.client.ErpRealClient;
import com.optivem.shop.dsl.driver.adapter.external.erp.client.dtos.ExtCreateProductRequest;
import com.optivem.shop.dsl.driver.port.external.erp.dtos.ReturnsProductRequest;
import com.optivem.shop.dsl.driver.port.external.erp.dtos.ReturnsPromotionRequest;
import com.optivem.shop.dsl.driver.port.shared.dtos.ErrorResponse;
import com.optivem.shop.dsl.common.Result;

public class ErpRealDriver extends BaseErpDriver<ErpRealClient> {
    private static final String DEFAULT_TITLE = "Test Product Title";
    private static final String DEFAULT_DESCRIPTION = "Test Product Description";
    private static final String DEFAULT_CATEGORY = "Test Category";
    private static final String DEFAULT_BRAND = "Test Brand";

    public ErpRealDriver(String baseUrl) {
        super(new ErpRealClient(baseUrl));
    }

    @Override
    public Result<Void, ErrorResponse> returnsPromotion(ReturnsPromotionRequest request) {
        return Result.success();
    }

    @Override
    public Result<Void, ErrorResponse> returnsProduct(ReturnsProductRequest request) {
        var createProductRequest = ExtCreateProductRequest.builder()
                .id(request.getSku())
                .title(DEFAULT_TITLE)
                .description(DEFAULT_DESCRIPTION)
                .category(DEFAULT_CATEGORY)
                .brand(DEFAULT_BRAND)
                .price(request.getPrice())
                .build();

        return client.createProduct(createProductRequest)
                .mapError(ext -> ErrorResponse.builder().message(ext.getMessage()).build());
    }
}
